src/Features/CSharp/Portable/Snippets/CSharpSnippetFunctionService.cs

```diff
@@ -0,0 +1,74 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Composition;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.CSharp;
+
+[ExportLanguageService(typeof(SnippetFunctionService), LanguageNames.CSharp), Shared]
+internal class CSharpSnippetFunctionService : SnippetFunctionService
```

Comment:
extracted this code from the c# snippet functions
https://github.com/dotnet/roslyn/tree/main/src/VisualStudio/CSharp/Impl/Snippets/SnippetFunctions